pkg/cmd/roachtest/tests/mvcc_gc.go

```diff
@@ -0,0 +1,443 @@
+		values[i] = fmt.Sprintf("(%d)", getSplitKey(i+1, ranges))
+	}
+	stmt := fmt.Sprintf(`alter table %s.%s split at values %s with expiration now() + INTERVAL '%d'`,
+		databaseName, tableName, strings.Join(values, ", "), int(expiration.Seconds()))
+	if _, err := conn.Exec(stmt); err != nil {
+		t.Fatalf("failed to split kv table into ranges: %s", err)
+	}
+}
+
+// Get split point splits uint64 key space into 'fragments' chunks and returns
+// 'index' chunk start.
+func getSplitPoint(index, fragments int) uint64 {
+	return math.MaxUint64 / uint64(fragments) * uint64(index)
+}
+
+// Get split keys gives back int64 split points for keys used by kv workload.
+func getSplitKey(index, fragments int) int64 {
+	return math.MinInt64 + int64(getSplitPoint(index, fragments))
+}
+
+func assertTableHasNoGarbage(t test.Test, conn *gosql.DB, tableName, databaseName string) error {
+	totals, rangeCount, sts := collectTableMVCCStatsOrFatal(t, conn, tableName, databaseName)
+	t.L().Printf("assert range tombstones and garbage in ranges: %d stats: %s", rangeCount, totals.String())
+	if rangeCount == 0 {
+		return errors.New("failed to find any ranges for table")
+	}
+	if totals.RangeKeyCount > 0 || totals.RangeKeyBytes > 0 {
+		return errors.Errorf("table ranges contain range tombstones %s", totals.String())
+	}
+	if totals.GCBytesAge != 0 || totals.GCBytes() > 0 {
+		return errors.Errorf("table ranges contain garbage %s", totals.String())
+	}
+	if c := sts[roachpb.CheckConsistencyResponse_RANGE_INCONSISTENT.String()]; c > 0 {
+		return errors.Errorf("consistency check reported %d inconsistent ranges", c)
+	}
+	return nil
+}
+
+func checkTableHasNoLiveData(t test.Test, conn *gosql.DB, tableName, databaseName string) error {
+	totals, rangeCount, _ := collectTableMVCCStatsOrFatal(t, conn, tableName, databaseName)
+	t.L().Printf("assert live data in ranges: %d stats: %s", rangeCount, totals.String())
+	if rangeCount == 0 {
+		return errors.New("failed to find any ranges for table")
+	}
+	if totals.LiveBytes > 0 || totals.LiveCount > 0 ||
+		totals.IntentBytes > 0 || totals.IntentCount > 0 || totals.SeparatedIntentCount > 0 {
+		return errors.Errorf("table ranges contain live data %s", totals.String())
+	}
+	return nil
+}
+
+// rangeConsistencies maps consistency responses to how many times they were
+// found across table ranges.
+type rangeConsistencies map[string]int
+
+func collectTableMVCCStatsOrFatal(
+	t test.Test, conn *gosql.DB, tableName, databaseName string,
+) (enginepb.MVCCStats, int, rangeConsistencies) {
+	var startKey, endKey roachpb.Key
+	queryRowOrFatal(t, conn,
+		`select min(start_key), max(end_key) from crdb_internal.ranges_no_leases where database_name = $1 and table_name = $2`,
+		[]interface{}{tableName, databaseName}, []interface{}{&startKey, &endKey})
+
+	rows, err := conn.Query(`select range_id, start_key, status, crdb_internal.range_stats(start_key) from crdb_internal.check_consistency(false, $1, $2) order by start_key`,
```

Comment:
We should grab and output `detail` as well, since it will tell us what was wrong in the stats.